### PR TITLE
Testbench UUID

### PR DESCRIPTION
**Pull request by juimonen**
Initial code to enable .so loading based on UUID from topology.

I'm not sure how I could re-use the existing UUID definitions, here they 
are copied as global structs for testbench (also as CONFIG_LIBRARY is defined 
when compiling testbench and the uuid macros are empty).

Anyway, as example crossover uuid definition is added, actually not sure 
about difference between crossover and dcblocker (?). I tested the code with 
EQ uuid by setting the old enum definition in library_table to SOF_COMP_NONE.

### CODE REVIEW

File: tools/testbench/topology.c [hunk 1]
```diff
@@ -64,8 +64,11 @@ void register_comp(int comp_type)
 
 	/* get index of comp in shared library table */
 	index = get_index_by_type(comp_type, lib_table);
-	if (index < 0)
-		return;
+	if (index < 0 && comp_ext) {
```
Comment (singalsu) [inline tools/testbench/topology.c:67]: Instead of "index < 0" check here "comp_type == SOF_COMP_NONE"

Comment (juimonen) [reply tools/testbench/topology.c:67]: oh yes, thanks!

File: tools/testbench/topology.c [hunk 2]
```diff
@@ -614,15 +617,22 @@ int load_process(void *dev, int comp_id, int pipeline_id,
...
-	ret = tplg_load_process(comp_id, pipeline_id, size, &process, file);
+	ret = tplg_load_process(comp_id, pipeline_id, size, &process, file, &comp_ext);
 	if (ret < 0)
 		return ret;
 
+	printf("debug uuid:\n");
+	for (i = 0; i < SOF_UUID_SIZE; i++)
+		printf("%u ", comp_ext.uuid[i]);
+	printf("\n");
+
```
Comment (singalsu) [inline tools/testbench/topology.c:635]: You need to remove the next "if (process.comp.type == SOF_COMP_NONE)" test to pass with non-legacy component.

Comment (juimonen) [reply tools/testbench/topology.c:635]: yes, sorry was testing only with modified iir, will change.

**Comment by singalsu** [comment]
Thanks @juimonen ! With those small modifications I was able to crossover and beamformer.

**Review by singalsu — APPROVED** [review]
This works for me, thanks Jaska!

Note: To add component there's need to increment NUM_WIDGETS_SUPPORTED, add DECLARE_SOF_TB_UUID(), add component to lib_table[].

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Looks like we had a kernel error on Jenkins CML. Rerun CI - @juimonen is this still RFC ? Can we merge once CI is good ?

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by juimonen** [comment]
@lgirdwood from my point of view we can integrate. @singalsuo got at least beamformer working with this, not sure about others...

**Pull request merged**